stackdriver-errors.js

```diff
@@ -179,6 +179,15 @@ function sendErrorPayload(url, payload) {
         var code = xhr.status;
         if (code >= 200 && code < 300) {
           resolve({message: payload.message});
+        } else if (code === 429) {
```

Comment:
Feedback desired:
It felt unnatural to not reject the promise in some way when the response is a 429, so I'm rejecting it with an object that has the same ['required' properties as `Error`](http://go/mdn/JavaScript/Reference/Global_Objects/Error#instance_properties). In this way, if someone did:
```
reporter
    .report(...)
    .catch((error) => { ...do something else with error.message or error.name...  })
;
```

It would meet minimum expectations. If you'd prefer, I could just reject `undefined` or not reject at all.

I don't think we run into an issue by rejecting with an object, since `reportUnhandledPromiseRejections` seems to [catch with a noop function](https://github.com/GoogleCloudPlatform/stackdriver-errors-js/blob/master/stackdriver-errors.js#L68), and therefore wouldn't be unhandled.

Reply:
Good idea